### PR TITLE
Add more features for D language

### PR DESCRIPTION
**Pull request by tuhdo**
Add `company-dcd` package that uses DCD (a completion daemon for D) to
provide better IDE features:

- Better code completion.
- Jump to definition.
- Search symbol references.

### CODE REVIEW

**Comment by tuhdo** [comment]
Some demos:

dcd-complete1

dcd-complete2

File: layers/+lang/d/packages.el [hunk 1]
```diff
@@ -26,6 +27,18 @@
   (with-eval-after-load 'company-dabbrev-code (push 'd-mode company-dabbrev-code-modes))
   (spacemacs|add-company-hook d-mode))
 
+(defun d/init-company-dcd ()
+  (use-package company-dcd
```
Comment (bmag) [inline layers/+lang/d/packages.el]: Should probably be lazy-loaded (`:defer t`)

Comment (tuhdo) [inline layers/+lang/d/packages.el]: @bmag done.

File: layers/+lang/d/packages.el [hunk 2]
```diff
@@ -14,6 +14,7 @@
 (setq d-packages
       '(
         company
+        company-dcd
```
Comment (TheBB) [inline layers/+lang/d/packages.el]: Add `:toggle (configuration-layer/package-usedp 'company)` here and it looks good to me.

Comment (tuhdo) [inline layers/+lang/d/packages.el]: @TheBB Done.

**Comment by trishume** [comment]
Oh man, improving Nim and D! Trying out some cool new languages?

**Comment by tuhdo** [comment]
@trishume yea, I am checking them out :D. Currently, I'm in favor of Nim.

**Comment by d12frosted** [comment]
Thank you :+1: Cherry-picked into `develop`, you can safely delete your branch.

**Pull request merged**